### ISSUE TITLE
Hour Account  > Minus hours calculations? > Not Approved OT Hours still shown

### ISSUE THREAD

**Issue by borishinzer** [horilla-opensource/horilla#31]
Is there a way to see a total summary of either overtimes or minus hours per day, week, month - and also transfer those to the next month or to reset them, in case overtimes were paid?

Also, similar to overtimes, minus hours should be reported and approved by a supervisor.

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Currently we do not have this feature added in our system.

**Comment by borishinzer** [comment]
Any update on this feature? IMO it would only be showing either overtimes or undertimes (minus hours) in hour account.

**Comment by CybroOdooDev** [comment]
Hi @borishinzer ,
Not presently.
We have it in our queue.
Sorry for the delay.
With Regards,
** Team Horilla **

**Comment by borishinzer** [comment]
Any update on this issue? This would be really awesome to integration as it would be our final issue (and fix for https://github.com/horilla-opensource/horilla/issues/54)  to move completely to Horilla in favour of Kimai, which we used formerly.

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
We get your concern.
We'll get back to you with the solution to this by the end of this week.
Sorry for the delay to this.

With Regards,
**Team Horilla**

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
We have updated the hour calculation method.
Please check the [commit](https://github.com/horilla-opensource/horilla/commit/e8c7eddd4c8b7c90a0c94e4b935128c043552d9a)
Please refer the video for the workflow.

https://github.com/horilla-opensource/horilla/assets/131998600/e5e11082-36fe-4eed-a6f3-3a67375f445e

With Regards,
**Team Horilla**

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Any updates on the above solution?

With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
Do I get this right? 
Pending hours are "minus hours" (if the employee worked too few hours in one month / period).
Also, if the times are validated then they are taking into account, resulting in either overtimes or "pending hours"?

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Yes,  pending hours are considered as minus hours.
If attendance with pending hours are validated, then it is considered for the monthly account.
Similarly in the case of overtime. Only once approved, it'll be added to the accounts.

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
We have added the column for viewing the pending hours in My Attendances.
<img width="1470" alt="Screenshot 2023-12-01 at 3 56 01 PM" src="https://github.com/horilla-opensource/horilla/assets/131998600/c5dfbd00-6ddd-4747-b9b8-9e275adee441">

Please check.

With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
Awesome. Checking now our time tracking for November and will update the issue here.

**Comment by borishinzer** [comment]
Seems last update broke the validation somehow. I try to validate various entries, but they stay for validation.

**Comment by CybroOdooDev** [comment]
Are you getting any kind of error notifications or  is it its just not getting validated even after clicking the validate button?

**Comment by borishinzer** [comment]
Tried in Firefox incognito mode - issue stays the same. Where to find the logs?

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Sorry for the delay.
We were able to reproduce the issue mentioned above from our side and we have made a fix for it.

Please check: [[FIX] ATTENDANCE: Attendance validation issue ](https://github.com/horilla-opensource/horilla/commit/36e4a922a99c32350ac93ac972e6e956e94fcb34)

The issue was with the validation methods.
There was an existing request pending for that corresponding attendance. So on validating it was causing the issue.
In the new workflow user will be given a warning if there is an existing request pending prior to validation and user can view the corresponding request from that view where they will be able to approve it.

<img width="1470" alt="Screenshot 2023-12-02 at 2 22 09 PM" src="https://github.com/horilla-opensource/horilla/assets/131998600/2f2295bd-a5d5-4519-825d-2f342947c37a">


With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
You can find a video of the issue here: https://cloud.web-vision.de/index.php/s/oN7RMiNHXP4Fz7H

**Comment by borishinzer** [comment]
Additional: it seems that sometime "approval" is working and sometimes not.

**Comment by horilla-opensource** [comment]
Sure, we'll check and update the solution asap.

**Comment by CybroOdooDev** [comment]
> Still not completely fixed. I get the request and then head over to "approve" the time entry, which doesn't change anything. If I "edit" and "save" the record without changing anything, it will get automatically approved.

Hi @borishinzer ,
Could you check with the latest commits for the fix?

**Comment by borishinzer** [comment]
We didn't validate the time calculations yet, as we needed a full month of time trackings before. Our people  are now actively testing Horilla based on the time trackings for November, December. Our aim is to report bugs and improvements and start using Horilla only (as substitute for Kimai) with beginning of November.

At this point I would like to says THANK YOU for your incredible work and your fast responses. This is really awesome 🙏

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Thanks for taking your time in testing out Horilla and providing us with your suggestions to improve our system.
We are ready to assist you in all the support needed for setting up Horilla in your side.
Please feel free to contact us with any queries. (info@horilla.com / nikhil@cybrosys.in)

With Regards,
**Team Horilla**

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
Any  updates on the issue fix?

With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
Still not working properly. I have validated all times of an employee and also approved his overtimes. Still the hour account shows wrong values and pending approvals.
<img width="1919" alt="Bildschirmfoto 2024-03-28 um 10 39 00" src="https://github.com/horilla-opensource/horilla/assets/10340538/08846965-72b3-4f6d-a65d-573cff6776f6">

**Comment by borishinzer** [comment]
Any updates on this task?

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
We have made changes for the calculation methods in our previous commits. Can you confirm with the latest commits if its fixed for you ?

We apologize for not providing an update on the same.

With Regards,
**Team Horilla**

**Comment by horilla-opensource** [comment]
Hi @Riiiad ,
Can you confirm in the overtime tab whether it is need to be approved?

In the attendance menu, you can find _OT Attendances_, please check whether there is any overtime pending for this employee that month.
<img width="1430" alt="Screenshot 2024-09-26 at 1 55 00 PM" src="https://github.com/user-attachments/assets/d7a831a4-ffcd-4b62-b380-ef0b7091f43e">

With Regards,
**Team Horilla**

**Comment by Riiiad** [comment]
I can confirm that all times are Approved:

<img width="1551" alt="Screenshot 2024-09-26 at 11 21 24" src="https://github.com/user-attachments/assets/56d59b11-df06-46c7-9fe4-de04bba796ed">

**Comment by horilla-opensource** [comment]
Hi @Riiiad ,
Can you please share the _**Attendance Break Point**_ settings inside the _Settings_ ?
<img width="1383" alt="Screenshot 2024-09-26 at 3 12 06 PM" src="https://github.com/user-attachments/assets/079f1c80-a071-4317-ae21-30ea90c3593c">

Also can you please share the details of the _**Not Approved OT Hours**_ column in the _**Hour Account**_ view (marked in red in the screenshot)
<img width="1421" alt="Screenshot 2024-09-26 at 3 13 39 PM" src="https://github.com/user-attachments/assets/922cfcf2-d903-4f25-9d5a-7af32cb620fb">

_In case if you don't see this column, you can enable the column by clicking in the hamburger menu dot icon (marked green in the screenshot) and enabling that column._

**Comment by Riiiad** [comment]
I dont undestand why there is 2 Places to approve overtimes?
OT Tab and Hour Account... this is confusing UI
I dont have the option Attendance Break point dont know show to get there.

**Comment by Riiiad** [comment]
When I Confirm the Overtime in the Hour Account also it shows in the View...
The problem is I have to approve everyday even if it has no overtime. Is there a Bulk option to Confirm all days or at lease a month?

**Comment by horilla-opensource** [comment]
Hi @Riiiad ,
 The screenshot I have added above is from the **_Hour Account_**. From the screenshot you have added, its of the Attendance view.

The **_Attendance Break Point_** can be fetched from the **_Settings_** option.
<img width="1429" alt="Screenshot 2024-09-26 at 3 42 19 PM" src="https://github.com/user-attachments/assets/09d7a16d-9a77-4332-9f8d-cd247ba6700a">

**Comment by horilla-opensource** [comment]
> When I Confirm the Overtime in the Hour Account also it shows in the View... The problem is I have to approve everyday even if it has no overtime. Is there a Bulk option to Confirm all days or at lease a month?

Yes, we do have an option for Bulk approval.
<img width="1388" alt="Screenshot 2024-09-26 at 3 45 30 PM" src="https://github.com/user-attachments/assets/1c9151ec-5d44-4065-ac70-3c41ee1971fc">

For Bulk Approval, select the records which you want to approve in bulk (as seen on 1 in the screenshot)

Once selected, click on the _**Actions**_ tab (maked as 2 in the screenshot )

Click on **_Approve OT_** and confirm.

**Comment by Riiiad** [comment]
No this is the wrong view you have. I need to Confirm in the Hour Account Tab. As in the OT Tab these entries are not there.

**Comment by horilla-opensource** [comment]
Can I know which version of Horilla you are using or the git head?

**Comment by Riiiad** [comment]
I need to check the exact commit but the Date is 24.September

**Comment by horilla-opensource** [comment]
Then it should be having the view that I have added in the screenshot above. 
Please confirm with the url from the screenshot also that we are in the same menu option view.

**Comment by Riiiad** [comment]
In the view you have send me I dont have any OT. /attendance/attendance-view/
Here:
<img width="1578" alt="Screenshot 2024-09-26 at 12 50 38" src="https://github.com/user-attachments/assets/20019359-814f-46c1-88e6-7e8b0d2b9d5c">

But here i have. attendance/attendance-overtime-view/
<img width="1556" alt="Screenshot 2024-09-26 at 12 51 43" src="https://github.com/user-attachments/assets/e3327061-0438-4083-b6f7-8ca34e1cd9a7">

**Comment by Riiiad** [comment]
And in the second there is not Bulk Option

**Comment by horilla-opensource** [comment]
Hi @Riiiad ,
The screenshot you have added in the previous comment is a bit confusing. The first screenshot with the url is correct.

But the screenshot and url given doesn't match. The second url (_attendance/attendance-overtime-view/_) you have mentioned is of _Hour Account_, but the screenshot is _Attendance_ view. 


_PS: I think there is confusion with the terms and urls that are currently in the view. We apologize for that confusion. We are updating that section in our next version_

**Comment by Riiiad** [comment]
Here is the Path you need to take to get to the confirm button:
https://github.com/user-attachments/assets/700d761b-f2d0-4a7f-9539-7e614d4d69e3

**Comment by horilla-opensource** [comment]
Can you share the _Attendance Break Point_ settings screenshot? 

> Hi @Riiiad , Can you please share the _**Attendance Break Point**_ settings inside the _Settings_ ? <img alt="Screenshot 2024-09-26 at 3 12 06 PM" width="1383" src="https://private-user-images.githubusercontent.com/131998600/371079854-079f1c80-a071-4317-ae21-30ea90c3593c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNTA1ODIsIm5iZiI6MTcyNzM1MDI4MiwicGF0aCI6Ii8xMzE5OTg2MDAvMzcxMDc5ODU0LTA3OWYxYzgwLWEwNzEtNDMxNy1hZTIxLTMwZWE5MGMzNTkzYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNlQxMTMxMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjFiYTY3MzZjNGQzMjAwMjkyNjY3OWEyYTgzODY5ZTFiOWNjY2FmNjMzYWVjMjdmOWM4NzI2NWUwYTk0MmY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2ve8PAovlfYFfBnwksJaYMRm-O68wHZVJkcSgncV0jQ">
> 
> Also can you please share the details of the _**Not Approved OT Hours**_ column in the _**Hour Account**_ view (marked in red in the screenshot) <img alt="Screenshot 2024-09-26 at 3 13 39 PM" width="1421" src="https://private-user-images.githubusercontent.com/131998600/371080696-922cfcf2-d903-4f25-9d5a-7af32cb620fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNTA1ODIsIm5iZiI6MTcyNzM1MDI4MiwicGF0aCI6Ii8xMzE5OTg2MDAvMzcxMDgwNjk2LTkyMmNmY2YyLWQ5MDMtNGYyNS05ZDVhLTdhZjMyY2I2MjBmYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNlQxMTMxMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYzM2YjczMjk0YThiM2ExMzU1ZTVkYTQ4ZmZkMDkxNGFiYmVlMjhhMDYxNDIxM2FjYTUwMWZkODgwMDBmNDk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5pKp-CC-w1TkJZ6diqggtWiqaQnM26Hz89HLC5UWMBA">
> 
> _In case if you don't see this column, you can enable the column by clicking in the hamburger menu dot icon (marked green in the screenshot) and enabling that column._

**Comment by horilla-opensource** [comment]
Hi @Riiiad ,
From the Attendance Break Point condition you have setup, you need to have a minimum of 01:30 to have an overtime. But from the overtimes you have shared, there is no record which satisfy the minimum hour condition. That's why its not being displayed.

Please change the minimum hour for over time to a lower value and you'll be able to see the record inside the OT tab.

**Comment by Riiiad** [comment]
@horilla-opensource But that means that there is not Auto confirm is that correct. I have to Confirm even 1min OT.

**Comment by horilla-opensource** [comment]
Yes, in the current version there is no auto confirm.
But the work is in progress for the auto confirm of overtime, just like the attendance validation which will soon be added to the system.

**Comment by borishinzer** [comment]
Which right does an employee need to confirm his/her overtimes (bulk) ?

**Comment by horilla-opensource** [comment]
> Which right does an employee need to confirm his/her overtimes (bulk) ?

Hi @borishinzer 
In the current version, the reporting manager and HR admin has the access to approve the over time of the employees(subordinates).

We are making an update for auto approval for overtime based on a minimum condition as given for the attendance validations.

With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
@horilla-opensource , thanks for the feedback, but in the meantime we want to make sure that at least the teamleads can approve times of employees. But we can't find the right setting.

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,
The teamleads are considered as reporting manager in Horilla. So if an employee has the need for approval in any form, his/her reporting manager has the access to approve that.

Also for the automatic overtime approval, we have added option inside the attendance breakpoint condition. ([[ADD] ATTENDANCE: Auto approve overtime](https://github.com/horilla-opensource/horilla/commit/40440eff0ec1dd74a5f44003f9eb0ab198b9eeff))

<img width="608" alt="Screenshot 2024-09-30 at 2 41 27 PM" src="https://github.com/user-attachments/assets/33c67296-4fb8-4610-8506-3a01e81ad33b">

With Regards,
**Team Horilla**

**Comment by borishinzer** [comment]
Thanks for the update. One question: the setting "Minimum Hour to Approve Overtime" is meant to be the auto-approval threshold?

**Comment by horilla-opensource** [comment]
Yes, once the employee's overtime is above the _Minimum Hour to Approve Overtime_, the overtime will be automatically approved and does not require manual approval of their managers.

**Comment by borishinzer** [comment]
The auto-approval feature is only "hiding" entries from the confirmation, but not calculating them into the total hour-acount. Example: We set the threshold to 1:00 hour for auto-approval. And every overtime below 1h was simply missing (gone). Once we set it back to 0:00 hour and confirming it then, it was reflecting in the correct amount of hours in the hour account.

**Comment by horilla-opensource** [comment]
OT attendances only listed in the Overtime Tab when the attendance's OT complete minimum hours to approve OT hours. if you enable auto approve feature then all the attendance satisfy this minimum conditions get approved automatically.

**Comment by borishinzer** [comment]
The issue is **not** that they are no longer visible. The issue is that if you put a threshold (eg. 1h) every OT below this threshold will not taken into calculation of the Overtimes/Pending Hours.

**Comment by horilla-opensource** [comment]
Hi @borishinzer ,

In the new update, all the OT attendances are listed in the Ot attendance tab. if the attendance overtime do not achieve the minimum OT condition then while approve the attendance a confirmation is asked, so weather you can decide to approve or not for those attendance. and the automated approve is work for those they satisfy the minimum condition.

**Comment by horilla-opensource** [comment]
Hi @,
We are closing this issue due to inactivity. It seems that there hasn't been any recent activity or discussion, and we believe it may have been resolved or is no longer relevant. 
If you feel this issue is still important and should be addressed, please feel free to reopen it or create a new issue with updated details. 

With Regards,
**Team Horilla**
